sematic/resolvers/log_streamer.py

```diff
@@ -87,21 +90,33 @@ def _flush_to_file(
             os.remove(file_path)
 
     started_reading = time.time()
+    received_stream_termination = False
 
     # Use w+ mode; should overwrite whatever was in the prior delta file
     with open(file_path, "w+") as fp:
         while (
             timeout_seconds is None or time.time() - started_reading < timeout_seconds
         ):
             line = read_handle.readline()
-            if len(line) == 0:
+            if _TERMINATION_CHAR in line:
```

Comment:
Had to do this so I could stop using "nothing to read" as both a signal for flushing and a signal for possible end-of-stream. Using it as a signal for flushing led to far too many flushes.